### PR TITLE
patching GetSignedRequestSignature to cover edge cases with the signature

### PR DESCRIPTION
**Pull request by RanVaknin**
This PR Addresses : https://github.com/aws/aws-sdk-go-v2/issues/2830

`GetSignedRequestSignature` function was changed to handle edge cases with signature parsing in the authorization header. It updates the string splitting logic to account for potential whitespace variations and adds test cases.

### CODE REVIEW

File: .changelog/f63b26b19e2e4330b4240f38088453ea.json [hunk 1]
```diff
@@ -0,0 +1,8 @@
+{
+    "id": "f63b26b1-9e2e-4330-b424-0f38088453ea",
+    "type": "bugfix",
+    "description": "Adding logic to omit whitespaces in the signature, and test coverage",
```
Comment (lucix-aws) [inline .changelog/f63b26b19e2e4330b4240f38088453ea.json:4]: In what?

```suggestion
    "description": "Improve handling of whitespace (or lack thereof) in sigv4 GetSignedRequestSignature.",
```

**Review by lucix-aws — APPROVED** [review]
fix and ship
